core/test/gather/fetcher-test.js

```diff
@@ -90,8 +90,30 @@ describe('._fetchResourceOverProtocol', () => {
         resource: {success: true, httpStatusCode: 200, stream: '1'},
       });
 
-    const data = await fetcher._fetchResourceOverProtocol('https://example.com', {timeout: 500});
-    expect(data).toEqual({content: streamContents, status: 200});
+    const data = await fetcher._fetchResourceOverProtocol('https://example.com', {
+      timeout: 500, responseHeaders: []});
+    expect(data).toEqual({content: streamContents, headers: null, status: 200});
+  });
+
+  it('returns requested headers if present', async () => {
```

Comment:
Can you add a test case where `headers` is `{}`. Like when the network response returned headers but none of them were explicitly requested.